In [ ]:
import pandas as pd
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import matplotlib


In [ ]:
df = pd.read_csv("../data/data.csv")

EXPLORATION DES DONNEES

In [ ]:
df.head()

In [ ]:
df.describe()

In [ ]:
df.info()
df['age'] = df['age'].astype(int)

In [ ]:
for col in ["ever_married","work_type","Residence_type","smoking_status","gender"]:
    print(df[col].unique())


In [41]:
df.isna().sum()

id                     0
gender                 0
age                    0
hypertension           0
heart_disease          0
ever_married           0
work_type              0
Residence_type         0
avg_glucose_level      0
bmi                  201
smoking_status         0
stroke                 0
dtype: int64

On remplace les 201 valeurs manquante dans BMI par la médiane car ça n'affecte pas les valeurs extrèmes. 

In [44]:
df["bmi"].fillna(df["bmi"].median(), inplace=True)

id                   0
gender               0
age                  0
hypertension         0
heart_disease        0
ever_married         0
work_type            0
Residence_type       0
avg_glucose_level    0
bmi                  0
smoking_status       0
stroke               0
dtype: int64